### PR TITLE
Adding basic image generation targets.

### PR DESCRIPTION
**Pull request by cheftako**
Adding simple build targets to create proxy-server and proxy-agent images.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: gen clean certs build
+ifndef REGISTRY
+$(error REGISTRY is not defined)
+endif
+ifndef PROJECT_ID
+$(error PROJECT_ID is not defined)
+endif
```
Comment (mcrute) [inline Makefile]: Can you embed these checks into the targets that require them? The way this is written you'll have to specify these even if you have no need for them.

File: artifacts/images/agent-build.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+COPY pkg/    pkg/
+COPY cmd/    cmd/
+COPY proto/  proto/
+COPY vendor/ vendor/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o proxy-agent sigs.k8s.io/apiserver-network-proxy/cmd/agent
+
+# Copy the loader into a thin image
+FROM ubuntu:latest
```
Comment (mcrute) [inline artifacts/images/agent-build.Dockerfile]: I think these should work with `scratch` right?

File: Makefile (hunk 1 above)
Comment (cheftako) [reply Makefile]: Good point. Fixed.

File: artifacts/images/agent-build.Dockerfile (hunk 2 above)
Comment (cheftako) [reply artifacts/images/agent-build.Dockerfile]: Give it a try. At some point we may want an image with more tools in it.

**Pull request merged**